Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -1765,21 +1721,136 @@ open class NavigationMapView: UIView {
+            } else if belowSymbol {
+                // find the topmost non symbol layer for layerIdentifier in belowSymbolLayers.
+                if targetLayer == nil,
+                   layerInfo.type.rawValue != "symbol",
+                   let sourceLayer = mapView.mapboxMap.style.layerProperty(for: layerInfo.id, property: "source-layer").value as? String,
+                   !sourceLayer.isEmpty {
+                    targetLayer = layerInfo.id
+                }
+            } else if aboveRoadName {
+                // find the topmost road name label layer for layerIdentifier in arrowLayers.
+                if targetLayer == nil,
+                   layerInfo.id.contains("road-label"),
+                   mapView.mapboxMap.style.layerExists(withId: layerInfo.id) {
+                    targetLayer = layerInfo.id
+                }
+            } else {
+                // find the topmost layer for layerIdentifier in aboveSymbolLayers.
+                if targetLayer == nil,
+                   let sourceLayer = mapView.mapboxMap.style.layerProperty(for: layerInfo.id, property: "source-layer").value as? String,
+                   !sourceLayer.isEmpty {
+                    targetLayer = layerInfo.id
+                }
+            }
+        }
+        
+        guard let targetLayer = targetLayer else { return layerPosition }
+        guard let layerPosition = layerPosition else { return .above(targetLayer) }
+        
+        if belowSymbol {
+            // For layers should be below symbol layers.
+            if case let .below(sequenceLayer) = layerPosition, !belowSymbolLayers.contains(sequenceLayer) {
+                // If the sequenceLayer isn't in belowSymbolLayers, it's above symbol layer.
+                // So for the layerIdentifier, it should be put above the targetLayer, which is the topmost non symbol layer,
+                // but under the symbol layers.
+                return .above(targetLayer)
+            } else {
+                return layerPosition
+            }
+        } else if aboveRoadName {
+            // For layers should be above road name labels but below other symbol layers.
+            if case let .below(sequenceLayer) = layerPosition, aboveSymbolLayers.contains(sequenceLayer) {
+                // If the sequenceLayer is in aboveSymbolLayers, it's above all symbol layers.
+                // So for the layerIdentifier, it should be put above the targetLayer, which is the topmost road name symbol layer.
+                return .above(targetLayer)
+            } else if case let .above(sequenceLayer) = layerPosition, belowSymbolLayers.contains(sequenceLayer) {
+                // If the sequenceLayer is in belowSymbolLayers, it's below all symbol layers.
+                // So for the layerIdentifier, it should be put above the targetLayer, which is the topmost road name symbol layer.
+                return .above(targetLayer)
+            } else {
```

Comment:
For simplification this can be merged into one condition that uses `||` statement.